src/components/Translate/Translate.react.js

```diff
@@ -43,31 +29,23 @@ class Translate extends Component {
     'nl-NL': nl,
   };
 
-  componentDidMount() {
-    let arr,
-      file,
-      defaultPrefLanguage = this.state.defaultPrefLanguage;
-    let text = this.state.text;
+  const getTranslation = text => {
+    const defaultPrefLanguage = UserPreferencesStore.getPrefLang();
+    let translatedText;
+
     if (defaultPrefLanguage !== 'en-US') {
-      if (this.lang.hasOwnProperty(defaultPrefLanguage)) {
-        arr = Object.keys(this.lang[defaultPrefLanguage]);
-        file = this.lang[defaultPrefLanguage];
+      if (lang.hasOwnProperty(defaultPrefLanguage)) {
+        translatedText = lang[defaultPrefLanguage][text];
       } else {
-        arr = Object.keys(jp);
-        file = jp;
+        translatedText = jp[text];
```

Comment:
Does this mean, we default to japanese?

Reply:
@akshatnitd Yes sir, at least that is what the previous developer intended to do, so I just made it functional. If you want I can change it to English.

Reply:
Yes, that would be right

Reply:
@akshatnitd Made the changes you asked for. Thanks for your time, sir.